mlflow/java/client/src/main/java/org/mlflow/tracking/MlflowClient.java

```diff
@@ -0,0 +1,191 @@
+package org.mlflow.tracking;
+
+import org.apache.http.client.utils.URIBuilder;
+
+import org.mlflow.api.proto.Service.*;
+import org.mlflow.tracking.creds.BasicMlflowHostCreds;
+import org.mlflow.tracking.creds.MlflowHostCredsProvider;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * Client to an MLflow Tracking Sever.
+ */
+public class MlflowClient {
+  private final MlflowProtobufMapper mapper = new MlflowProtobufMapper();
+  private final MlflowHttpCaller httpCaller;
+
+  /** Returns a default client based on the MLFLOW_TRACKING_URI environment variable. */
+  public MlflowClient() {
+    this(getDefaultTrackingUri());
+  }
+
+  /** Instantiates a new client using the provided tracking uri. */
+  public MlflowClient(String trackingUri) {
+    this(getHostCredsProviderFromTrackingUri(trackingUri));
+  }
+
+  /**
+   * Creates a new MlflowClient. Users should prefer constructing ApiClients via
+   * {@link #MlflowClient()} ()} or {@link #MlflowClient(String)} if possible.
+   */
+  public MlflowClient(MlflowHostCredsProvider hostCredsProvider) {
+    httpCaller = new MlflowHttpCaller(hostCredsProvider);
+  }
+
+  public Run getRun(String runUuid) {
+    URIBuilder builder = newURIBuilder("runs/get").setParameter("run_uuid", runUuid);
+    return mapper.toGetRunResponse(httpCaller.get(builder.toString())).getRun();
+  }
+
+  public RunInfo createRun(long experimentId, String sourceName) {
```

Comment:
I think we should also provide a version without sourceName that doesn't set anything here, because when you use `mlflow run`, we want it to inherit whatever that set. Will things freak out if we don't set a source name?

Also, I'd rename the parameter to appName instead of sourceName to make it a bit more obvious what this is.

Basically, we should have:
- createRun()
- createRun(experimentId)
- createRun(experimentId, sourceName)
- createRun(CreateRun)